### PR TITLE
test/osd: unit-tests for the scrubber scheduler

### PR DESCRIPTION
**Pull request by ronen-fr**
Adding unittest_scrub_sched, testing the way scrubs are scheduled.

Some minor changes to the scrub scheduler itself, cleaning up its
interfaces to facilitate mocking.

### CODE REVIEW

File: src/osd/scrubber/osd_scrub_sched.cc [hunk 1]
```diff
@@ -220,6 +219,48 @@ void ScrubQueue::update_job(ScrubJobRef scrub_job,
...
+  }
+
+  dout(15) << fmt::format(
+		": suggested: {} hist: {} v: {}/{} must: {} pool-min: {}",
+		res.proposed_time,
+		pg_info.history.last_scrub_stamp,
+		(bool)pg_info.stats.stats_invalid,
+		conf()->osd_scrub_invalid_stats,
+		(res.is_must == must_scrub_t::mandatory ? "y" : "n"),
+		res.min_interval)
```
Comment (NitzanMordhai) [inline src/osd/scrubber/osd_scrub_sched.cc:258]: are you missing 1 value here?

Comment (ronen-fr) [reply src/osd/scrubber/osd_scrub_sched.cc:258]: Don't think so. The dout() prefix in this file includes the __func__ (i.e. - the first
':', if that is what you meant, is after the func name)

Comment (NitzanMordhai) [reply src/osd/scrubber/osd_scrub_sched.cc:258]: got it

**Review by NitzanMordhai — APPROVED** [review]

**Review by neha-ojha — APPROVED** [review]

File: src/osd/scrubber/osd_scrub_sched.cc [hunk 2]
```diff
@@ -220,6 +219,48 @@ void ScrubQueue::update_job(ScrubJobRef scrub_job,
...
 }
 
+ScrubQueue::sched_params_t ScrubQueue::determine_scrub_time(
+  const requested_scrub_t& request_flags,
+  const pg_info_t& pg_info,
+  const pool_opts_t pool_conf) const
+{
+  ScrubQueue::sched_params_t res;
+
+  //dout(15) << fmt::format(": requested_scrub_t: {}", request_flags) << dendl; // RRR
```
[the remark this answers is not in the document]
Comment (ronen-fr) [reply src/osd/scrubber/osd_scrub_sched.cc]: fixed. Thanks

**Comment by ronen-fr** [comment]
Merged based on 'yuri7' and the specific Teuthology tests I've run.
See https://docs.google.com/document/d/1YeoFYoGZpdSJrSN7bXe9zDAC-G0GOqDBwZ_dMng9QcU/edit?usp=sharing for a summary of the 'official' yuri7 tests (no related failures).
My set of tests are http://pulpito.front.sepia.ceph.com/?branch=wip-rf-sched-test

**Pull request merged**